### PR TITLE
weather oracle

### PR DESCRIPTION
**Pull request by Retamogordo**

### CODE REVIEW

File: tutorials/substrate-inherents-tutorial/1-setup.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Introduction to Inherent Extrinsics
+A term extrinsic refers to a transaction in Substrate parlance.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/1-setup.mdx:2]:
```suggestion
The term _extrinsic_ refers to a transaction in Substrate parlance.
```

File: tutorials/substrate-inherents-tutorial/1-setup.mdx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+End users usually submit signed transactions to be executed and stored on a block chain.
+Substrate offers a functionality for submitting also unsigned extrinsics on chain.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/1-setup.mdx:4]:
```suggestion
Substrate offers a functionality for also submitting unsigned extrinsics on chain.
```

File: tutorials/substrate-inherents-tutorial/1-setup.mdx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+This kind of transaction is in use, for example, by Offchain Workers that usually implement buisiness logic requiring transaction execution.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/1-setup.mdx:5]:
```suggestion
This kind of transaction is in use, for example, by _offchain workers_ that usually implement business logic requiring transaction execution.
```

File: tutorials/substrate-inherents-tutorial/1-setup.mdx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+are not gossiped to the rest of the node for consensus, rather they are included into the block to be executed when the block is imported by the nodes.
+Inherent Extrinsics are used to include node-specific data into the the block it proposes.
+For instance, inherent extrinsics are used to include the node's local timestamp when it produces a block.
+In this tutorial your Client will provide custom data to the Runtime for inclusion into the block.
+
+## Setup
+You're going to leverage the standard Substrate node template where you'll 
+customize the Client side as well as develop a custom Runtime pallet for handling your inherent extrinsic.
+
+As a first step, please fork, clone and build [substrate-node-template](https://github.com/substrate-developer-hub/substrate-node-template)
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/1-setup.mdx:17]: Do you think users might need more info here? Is `cargo build --release` enough? Where can they find more help if it doesnt work is there a troubleshooting guide somewhere you could link?

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 5]
```diff
@@ -0,0 +1,155 @@
+# Client
+## Inherent Data Provider
+A block producing node relies on an ```InherentDataProvider``` whenever it is interested to pass
+data to Runtime upon authoring a new block.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:4]:
```suggestion
data to the runtime upon authoring a new block.
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+An ```InherentDataProvider``` trait implementor utilizes ```provide_inherent_data``` method to put data onto
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:5]:
```suggestion
An ```InherentDataProvider``` trait implementor utilizes the ```provide_inherent_data``` method to put data into
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 7]
```diff
@@ -0,0 +1,155 @@
+# Client
+## Inherent Data Provider
+A block producing node relies on an ```InherentDataProvider``` whenever it is interested to pass
+data to Runtime upon authoring a new block.
+An ```InherentDataProvider``` trait implementor utilizes ```provide_inherent_data``` method to put data onto 
+storage location under a key both the Client and the Runtime share.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:6]:
```suggestion
storage under a key both the Client and the Runtime share.
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+As the first step you'll define a struct implementing the aforementioned ```InherentDataProvider``` trait.
+
+Please head to ./node/src folder and add a file ```inherent_data_provider.rs``` with the following code:
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:10]:
```suggestion
Please head to `./node/src` folder and add a file ```inherent_data_provider.rs``` with the following code:
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+```
+which is the key under which the block producer passes inherent data to the Runtime. 
+
+The struct ```ExternalDataInherentProvider``` holds the data you want the block producer to pass to the Runtime
+and implements the trait ```InherentDataProvider``` by putting it onto storage from which the data will be then
+fetched and decoded by the Runtime.
+
+## Create Inherent Data Provider
+Now when your inherent data provider is ready it must be used by the block production.
+Inherent data provider instances are created by a closure passed to consensus block production module.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:77]:
```suggestion
Inherent data provider instances are created by a closure passed to the consensus block production module.
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+        );
+
+    Ok((slot, timestamp))
+},
+```
+code snippet.
+
+There are two identical snippets in the file, at the moment we are going to concentrate
+on that referred by ```sc_consensus_aura::start_aura```.
+We see that there are already two inherent data providers returned by the closure which
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:96]:
```suggestion
You can see that there are already two inherent data providers returned by the closure which
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+Aura uses to pass inherent data to the Runtime when it produces a block.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:97]:
```suggestion
AURA uses to pass inherent data to the Runtime when it produces a block.
```

File: tutorials/substrate-inherents-tutorial/2-client.mdx [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+Aura passes the current time and slot inherents to the Runtime.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/2-client.mdx:98]:
```suggestion
AURA passes the current time and slot inherents to the Runtime.
```

File: tutorials/substrate-inherents-tutorial/3-runtime.mdx [hunk 13]
```diff
@@ -0,0 +1,136 @@
+# Runtime
+Now it's a time to develop the second part of your Inherents project - the Runtime side.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/3-runtime.mdx:2]:
```suggestion
Now it's time to develop the second part of your Inherents project - the Runtime side.
```

File: tutorials/substrate-inherents-tutorial/5-weather-oracle.mdx [hunk 14]
```diff
@@ -0,0 +1,40 @@
+# (Bad) Weather Oracle
+## Warning
+This project by no means can be used as a legitimate example for a decentralized oracle design.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/5-weather-oracle.mdx:3]:
```suggestion
:::caution

This project by no means can be used as a legitimate example for a decentralized oracle design.

:::
```

File: tutorials/substrate-inherents-tutorial/5-weather-oracle.mdx [hunk 15]
```diff
@@ -0,0 +1,40 @@
+# (Bad) Weather Oracle
+## Warning
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/5-weather-oracle.mdx:2]:
```suggestion

```

File: tutorials/substrate-inherents-tutorial/5-weather-oracle.mdx [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+This project by no means can be used as a legitimate example for a decentralized oracle design.
+
+## Description
+This section describes a slightly more interesting use case of Inherents.
+Instead of providing a constant value the node will access a weather data provider endpoint which fetches current weather parameters at a provided geographical coordinates.
+A user submits a transaction were they order a weather report for a specified location.
+The node fetches the tranasction data from the last block via a custom (RuntimeAPI)[https://docs.substrate.io/reference/runtime-apis/]
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/5-weather-oracle.mdx:9]:
```suggestion
The node fetches the tranasction data from the last block via a custom [RuntimeAPI](https://docs.substrate.io/reference/runtime-apis/)
```

File: tutorials/substrate-inherents-tutorial/6-conclusion.mdx [hunk 17]
```diff
@@ -0,0 +1,14 @@
+# Conclusion
+
+- You have learned to pass a Node-specific data to the Runtime upon block production and create an inherent extrinsic call using that data.
```
Comment (niklasp) [inline tutorials/substrate-inherents-tutorial/6-conclusion.mdx:3]:
```suggestion
You have learned to pass a Node-specific data to the Runtime upon block production and create an inherent extrinsic call using that data.
```

File: yarn.lock
Comment (niklasp) [inline yarn.lock:1]: 🫣

**Review by niklasp — COMMENTED** [review]
thanks for that tutorial, only some minor changes, didnt test the code though so i trust you it works

🎉

File: tutorials/substrate-inherents-tutorial/1-setup.mdx (hunk 4 above)
Comment (Retamogordo) [reply tutorials/substrate-inherents-tutorial/1-setup.mdx:17]: It's practically the same procedure as described in the original template node Readme, no extra or different steps are required either for building or for node running

**Pull request closed without merging**